### PR TITLE
Multiple file analysis

### PR DESCRIPTION
**Pull request by marinaPauw**
Each file produces its own mzQC, and tsv files, with the name of the original mzml file in the name of the tsv file.

### CODE REVIEW

File: MzmlReader/MzmlReader.cs [hunk 1]
```diff
@@ -85,7 +87,12 @@ public Run LoadMzml(string path, double massTolerance, bool storeScansInMemory,
         public void ReadSourceFileMetaData(XmlReader reader, Run run)
         {
             bool cvParamsRead = false;
-            run.SourceFileName = reader.GetAttribute("name");
+            run.SourceFileName = Path.GetFileNameWithoutExtension(reader.GetAttribute("name"));
+            if (run.SourceFileName.ToLower().EndsWith(".wiff"))//in a .wiff.scan file you need to remove both extensions
```
Comment (PaulBrack) [inline MzmlReader/MzmlReader.cs]: Why are we doing this? I don't understand

File: MzmlReader/Run.cs [hunk 2]
```diff
@@ -27,6 +27,8 @@ public Run()
         public String FilePropertiesAccession;
         public String iRTpath;
         public List<IRTPeak> IRTPeaks { get; set; }
+        public double RunLowerBoundary { get; set; }
```
Comment (PaulBrack) [inline MzmlReader/Run.cs]: Is this mz boundaries? Is this MS1 or MS2? If it's MS2, we have this information already stored.

File: SwaMe/ChromatogramMetricGenerator.cs [hunk 3]
```diff
@@ -66,58 +63,72 @@ public void GenerateiRTChromatogram(Run run, double massTolerance)
...
-                        CalcDotProductParameters(yyy, length, irtpeak, dp, transSmoothInt);
-                        //Collect the peak metrics for each transition, which will then be averaged out for all transitions accross a possible peak
-                        sp = ProducePeakMetrics(starttimes, Smoothed, irtpeak, sp);
                        
                     }
 
                     sp.DotProduct = Math.Pow(dp.TiriPair, 2) / (dp.Tsquared * dp.Rsquared);
                     double RTscore = CalcRTScore(Count, run, sp);
-                    double score = 0.2 * sp.DotProduct + 0.05 *1/rank*pPeak.Alltransitions[0].Count() + 0.5 * RTscore;//The rank is a rank of the intensity as the peaks were sorted by intensity, the rank is simply the order it occurred in
+                    double score = 0.4 * sp.DotProduct+ 0.01 * pPeak.Alltransitions[0].Count()/(rank/10) + 0.4* RTscore;///The rank is a rank of the intensity as the peaks were sorted by intensity, the rank is simply the order it occurred in
```
Comment (PaulBrack) [inline SwaMe/ChromatogramMetricGenerator.cs]: Change in algorithm without any documentation

File: SwaMe/ChromatogramMetricGenerator.cs [hunk 4]
```diff
@@ -66,58 +63,72 @@ public void GenerateiRTChromatogram(Run run, double massTolerance)
...
                     if (highestScore < score)
                     {
                         highestScore = score;
                         irtpeak.RetentionTime = sp.RT;
                         irtpeak.FWHM = sp.FWHMAllTransitions / irtpeak.AssociatedTransitions.Count();//average fwhm
                         irtpeak.Peaksym = sp.PSAllTransitions / irtpeak.AssociatedTransitions.Count();
                     }
+
+                    logger.Info(Convert.ToString(sp.RT)+"-"+Convert.ToString(irtpeak.Mz)+"-"+Convert.ToString(RTscore));
```
Comment (PaulBrack) [inline SwaMe/ChromatogramMetricGenerator.cs]: Not sure this is a helpful message.

File: MzmlReader/MzmlReader.cs (hunk 1 above)
Comment (marinaPauw) [reply MzmlReader/MzmlReader.cs]: We had two variables, run.SourceFileName and run.SourceFileType. Previously we weren't using type, we simply had the full filename of the file from which the mzML was made with extension under run.SourceFileName, for example "myFile.wiff.scan". At the end when writing files, we are now incorporating the name of the original file into the name of the file we are writing so as to distinguish from the other files when running multiple files for example "metrics_myFile.json". So we can remove the filetype from the name at the point of writing the file, which was my original plan. But then I noticed we still had the run.SourceFileType variable that we weren't using so I decided to remove the filetype from the name at the beginning and put it into the variable run.SourceFileType as we do have other sections where we use the filetype.

**Pull request merged**